inline.h

```diff
@@ -3405,6 +3405,182 @@ S_lossless_NV_to_IV(const NV nv, IV *ivp)
     return FALSE;
 }
 
+/*
+ * S_iv_{add,sub,mul}_may_overflow(a, b, p) virtually compute "a <op> b"
+ * (where <op> is +, -, or *) in infinite precision, and, if the result
+ * is (or may be) not representable with IV, return true.
+ * Otherwise (no overflow), store the result to *p and return false.
+ * These functions allow false positives (so their names contain "may")
+ * to speed up simple common cases.
+ */
+
+/* Define IV_*_OVERFLOW_IS_EXPENSIVE below to nonzero value
+ * if strict overflow checks are too expensive
+ * (for example, for CPUs that has no hardware overflow detection flag).
```

Comment:
RISC V doesn't have a hardware overflow flag (or any classic carry, zero etc flags), so the compiler generates more complex code:
```
bool my_chk_add(long a, long b, long *result) {
    return __builtin_add_overflow(a, b, result);
}
; RISC-V
my_chk_add(long, long, long*):
        add     a5,a0,a1
        slt     a0,a5,a0
        slti    a1,a1,0
        sub     a0,a1,a0
        sd      a5,0(a2)
        snez    a0,a0
        ret
; amd64
"my_chk_add(long, long, long*)":
        add     rdi, rsi
        mov     QWORD PTR [rdx], rdi
        seto    al
        ret
; arm64
my_chk_add(long, long, long*):
        adds    x1, x0, x1
        str     x1, [x2]
        cset    w0, vs
        ret
```
(no action needed here)